src/views/connectionUI.ts

```diff
@@ -548,14 +562,16 @@ export class ConnectionUI {
 	public async promptForRetryCreateProfile(profile: IConnectionProfile, isFirewallError: boolean = false): Promise<IConnectionProfile> {
 		// Ask if the user would like to fix the profile
 		let errorMessage = isFirewallError ? LocalizedConstants.msgPromptRetryFirewallRuleAdded : LocalizedConstants.msgPromptRetryCreateProfile;
-		return this._vscodeWrapper.showErrorMessage(errorMessage, LocalizedConstants.retryLabel).then(result => {
-			if (result === LocalizedConstants.retryLabel) {
-				return ConnectionProfile.createProfile(this._prompter, this._connectionStore, this._context,
-					this.connectionManager.azureController, this._accountStore, profile);
-			} else {
-				return undefined;
-			}
-		});
+		let result = await this._vscodeWrapper.showErrorMessage(errorMessage, LocalizedConstants.retryLabel);
+		if (result === LocalizedConstants.retryLabel) {
+			return ConnectionProfile.createProfile(this._prompter, this._connectionStore, this._context,
+				this.connectionManager.azureController, this._accountStore, profile);
+		} else {
+			// user cancelled the prompt - throw error so that we know user cancelled
+			return new Promise((_, reject) => {
```

Comment:
`throw new CancelError()`

(do you really need a message here? If there's a use for it I guess, but typically the messages aren't something we use except to display to the user)